server/src/lib/server.rs

```diff
@@ -561,37 +553,10 @@ async fn instance_ensure(
 
 #[endpoint {
     method = GET,
-    path = "/instances/{instance_id}/uuid",
-    unpublished = true,
-}]
-async fn instance_get_uuid(
-    rqctx: Arc<RequestContext<Context>>,
-    path_params: Path<api::InstanceNameParams>,
-) -> Result<HttpResponseOk<Uuid>, HttpError> {
-    let context = rqctx.context().context.lock().await;
-
-    let context = context.as_ref().ok_or_else(|| {
-        HttpError::for_internal_error(
-            "Server not initialized (no instance)".to_string(),
-        )
-    })?;
-
-    if path_params.into_inner().instance_id != context.properties.name {
-        return Err(HttpError::for_internal_error(
-            "Instance name mismatch (path did not match struct)".to_string(),
-        ));
-    }
```

Comment:
This use of the hostname was somewhat odd - basically, we used the client-provided hostname here, for a path which should only be accessible to the control plane.

I get why we used it - it acted as a convenient way to get the redundant UUID - but we should probably avoid referring to the instance by hostname for internal purposes, as that's could theoretically change.

Regardless, that problem no longer exists after this PR.

Reply:
Side-note, I was definitely confused by seeing `path = ... {instance_id}`, when the variable is actually mean to be a string-based hostname.